src/lib/routing-tables.js

```diff
@@ -0,0 +1,219 @@
+'use strict'
+
+const _ = require('lodash')
+const routing = require('five-bells-routing')
+const ROUTE_EXPIRY = 45 * 1000 // milliseconds
+const SIMPLIFY_POINTS = 10
+
+class RoutingTables {
+  constructor (baseURI, localRoutes) {
+    this.baseURI = baseURI
+    this.sources = {} // { "sourceLedger" => routing.RoutingTable }
+    this.accounts = {} // { "connector;ledger" => accountURI }
+    this.addLocalRoutes(localRoutes)
+  }
+
+  /**
+   * @param {Routes} localRoutes - Each local route should include the optional
+   *   `destination_account` parameter. `connector` should always be `baseURI`.
+   */
+  addLocalRoutes (localRoutes) {
+    for (const localRoute of localRoutes) {
+      const table = this.sources[localRoute.source_ledger] ||
+        (this.sources[localRoute.source_ledger] = new routing.RoutingTable())
+      const route = new routing.Route(localRoute.points, {
+        minMessageWindow: localRoute.min_message_window,
+        nextLedger: localRoute.destination_ledger
+      })
+      table.addRoute(localRoute.destination_ledger, this.baseURI, route)
+    }
+    localRoutes.forEach(this.addRoute, this)
+  }
+
+  /**
+   * @param {Route} route
+   */
+  addRoute (route) {
+    const ledgerB = route.source_ledger
+    const ledgerC = route.destination_ledger
+    const connectorFromBToC = route.connector
+    const routeFromBToC = new routing.Route(route.points)
+
+    this.accounts[connectorFromBToC + ';' + ledgerB] = route.source_account
+    if (route.destination_account) {
+      this.accounts[connectorFromBToC + ';' + ledgerC] = route.destination_account
+    }
+
+    this.eachSource((tableFromA, ledgerA) => {
+      // Don't create A→B→A.
+      if (ledgerA === ledgerC) return
+      const routesFromAToB = tableFromA.destinations.get(ledgerB)
+      if (!routesFromAToB) return
+      // Don't create local route A→B→C if local route A→C already exists.
+      if (this.baseURI === connectorFromBToC && this._getLocalRoute(ledgerA, ledgerC)) return
+      const routeFromAToB = routesFromAToB.get(this.baseURI)
+      // Don't create A→C→B when there is no local A→C.
```

Comment:
Should this say `A->B->C`? I don't quite understand how this code has anything to do with creating a route `A→C→B`